src/pages/ScholarshipForm.jsx

```diff
@@ -1,21 +1,112 @@
 import React, { useState } from 'react';
 import { Container, Button, TextField } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
 import ScholarshipAmountField from '../components/ScholarshipAmountField';
 import DatePicker from '../components/DatePicker';
+import AmountType from '../types/AmountType';
+
+const useStyles = makeStyles({
+  textFieldStyle: {
+    minWidth: 400,
+  },
+});
+
+// regular expression used for valid website urls
+const websiteReg = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([-.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
+let amountHelperText = '';
 
 function ScholarshipForm() {
+  const classes = useStyles();
   const [formFieldStates, setFormFieldStates] = useState({
     name: '',
-    deadline: new Date(),
+    deadline: null,
     description: '',
-    amountType: '',
-    minAmount: 0,
-    maxAmount: 0,
+    amountType: null,
+    minAmount: '',
+    maxAmount: '',
     website: '',
     scholarshipType: '',
-    active: false,
   });
 
+  const [formFieldErrors, setFormFieldErrors] = useState({
+    name: false,
+    deadline: false,
+    description: false,
+    amountType: false,
+    website: false,
+    minAmount: false,
+    maxAmount: false,
+  });
+
+  // returns true if amount is invalid
+  function invalidAmountFields() {
```

Comment:
I feel like this validation doesn't belong here as it's specific to the `ScholarshipAmountField` component. You'd have to copy this logic over to another form if the `ScholarshipAmountField` component ever gets used elsewhere.

Reply:
Alright I put it into another file and import the function in the `ScholarshipForm` page. That grants us re-usability.